snooty/types.py

```diff
@@ -311,7 +312,15 @@ def render_constants(self) -> Tuple["ProjectConfig", List[Diagnostic]]:
 
     def read(self, path: Path) -> Tuple[str, List[Diagnostic]]:
         text = path.read_text(encoding="utf-8")
-        return self.substitute(text)
+        source_text, diagnostics = self.substitute(text)
+        match_found = PAT_GIT_MARKER.finditer(text)
+
+        if match_found:
+            for match in match_found:
+                lineno = source_text.count("\n", 0, match.start())
+                diagnostics.append(Diagnostic.error("git merge conflict found", lineno))
```

Comment:
Should the diagnostic error also report the file path to make it easier for the user to parse (and also thinking about standard automated log parsing tools)?

Reply:
Diagnostics are always associated with a `Page` instance that stores the path